kornia/geometry/bbox_v2.py

```diff
@@ -0,0 +1,588 @@
+    boxes = boxes if boxes.is_floating_point() else boxes.float()
+
+    xmin, ymin, zmin = boxes[..., 0], boxes[..., 1], boxes[..., 2]
+    if mode in ("xyzxyz", "xyzxyz_plus_1"):
+        width = boxes[..., 3] - boxes[..., 0]
+        height = boxes[..., 4] - boxes[..., 1]
+        depth = boxes[..., 5] - boxes[..., 2]
+    elif mode in ("xyzwhd", "xyzwhd_plus_1"):
+        depth, height, width = boxes[..., 4], boxes[..., 3], boxes[..., 5]
+    else:
+        raise ValueError(f"Unknown mode {mode}")
+
+    if "plus_1" in mode:
+        height -= 1
+        width -= 1
+        depth -= 1
+
+    hexahedrons = _boxes3d_to_polygons3d(xmin, ymin, zmin, width, height, depth)
+    hexahedrons = hexahedrons if batched else hexahedrons.squeeze(0)
+    return hexahedrons
+
+
+def kornia_bbox_to_bbox(kornia_boxes: torch.Tensor, mode: str = "xyxy") -> torch.Tensor:
+    r"""Convert 2D bounding boxes in kornia format to the format specified by ``mode``.
+
+    Args:
+        kornia_boxes: boxes to be transformed, shape of :math:`(N, 4, 2)` or :math:`(B, N, 4, 2)`.
+        mode: the ouput box format. It could be:
+
+            * 'xyxy': boxes are defined as ``xmin, ymin, xmax, ymax`` where ``width = xmax - xmin`` and
+              ``height = ymax - ymin``.
+            * 'xyxy_plus_1': like 'xyxy' where ``width = xmax - xmin + 1`` and  ``height = ymax - ymin + 1``.
+            * 'xywh': boxes are defined as ``xmin, ymin, width, height`` where ``width = xmax - xmin``
+              and ``height = ymax - ymin``.
+            * 'xywh_plus_1': like 'xywh' where ``width = xmax - xmin + 1`` and  ``height = ymax - ymin + 1``.
+            * 'vertices': boxes are defined by their vertices points in the following ``clockwise`` order: *top-left,
+              top-right, bottom-right, bottom-left*. Vertices coordinates are in (x,y) order. Finally, box width and
+              height are defined as ``width = xmax - xmin`` and ``height = ymax - ymin``.
+            * 'vertices_plus_1': like 'vertices' where ``width = xmax - xmin + 1`` and  ``height = ymax - ymin + 1``.
+
+    Returns:
+        Bounding boxes tensor the ``mode`` format. The shape is :math:`(N, 4, 2)` or :math:`(B, N, 4, 2)` in 'vertices'
+        or 'vertices_plus_1' mode and :math:`(N, 4)` or :math:`(B, N, 4)` in all other cases.
+
+    Examples:
+        >>> boxes_xyxy = torch.as_tensor([[0, 3, 1, 4], [5, 1, 8, 4]])
+        >>> kornia_bbox = bbox_to_kornia_bbox(boxes_xyxy, mode='xyxy')
+        >>> assert (kornia_bbox_to_bbox(kornia_bbox, mode='xyxy') == boxes_xyxy).all()
+    """
+    _check_bbox_dimensions(kornia_boxes)
+    batched = kornia_boxes.ndim == 4
+    # Create (B,N,4,2) or (N,4,2) with all points in top left position of the bounding box
+    kornia_boxes = kornia_boxes if batched else kornia_boxes.unsqueeze(0)
+
+    # Create boxes in xyxy format.
+    boxes = torch.stack([kornia_boxes.min(dim=-2).values, kornia_boxes.max(dim=-2).values], dim=-2).view(
```

Comment:
Would ".values" break the gradients?

Reply:
I don't know. Pytorch docs has [this](https://pytorch.org/docs/stable/generated/torch.min.html?highlight=min#torch.min) warning: `This function produces deterministic (sub)gradients unlike min(dim=0)`. Could it be "dim=-2" part?

Reply:
That warning probably related to https://github.com/pytorch/pytorch/issues/35699.